nucypher/blockchain/eth/utils.py

```diff
@@ -62,3 +69,107 @@ def get_tx_cost_data(transaction_dict: TxParams):
+    """
+    query = {
+        "jsonrpc": "2.0",
+        "method": "eth_getBlockByNumber",
+        "params": ["latest", False],
+        "id": 1,
+    }
+    LOGGER.debug(f"Checking health of RPC endpoint {endpoint}")
+    try:
+        response = requests.post(
+            endpoint,
+            json=query,
+            headers={"Content-Type": "application/json"},
+            timeout=5,
+        )
+    except requests.exceptions.RequestException:
+        LOGGER.debug(f"RPC endpoint {endpoint} is unhealthy: network error")
+        return False
+
+    if response.status_code != 200:
+        LOGGER.debug(
+            f"RPC endpoint {endpoint} is unhealthy: {response.status_code} | {response.text}"
+        )
+        return False
+
+    try:
+        data = response.json()
+        if "result" not in data:
+            LOGGER.debug(f"RPC endpoint {endpoint} is unhealthy: no response data")
+            return False
+    except requests.exceptions.RequestException:
+        LOGGER.debug(f"RPC endpoint {endpoint} is unhealthy: {response.text}")
+        return False
+
+    if data["result"] is None:
+        LOGGER.debug(f"RPC endpoint {endpoint} is unhealthy: no block data")
+        return False
+
+    block_data = data["result"]
+    try:
+        timestamp = int(block_data.get("timestamp"), 16)
+    except TypeError:
+        LOGGER.debug(f"RPC endpoint {endpoint} is unhealthy: invalid block data")
+        return False
+
+    system_time = time.time()
+    drift = abs(system_time - timestamp)
+    if drift > max_drift_seconds:
+        LOGGER.debug(
+            f"RPC endpoint {endpoint} is unhealthy: drift too large ({drift} seconds)"
+        )
+        return False
+
+    LOGGER.debug(f"RPC endpoint {endpoint} is healthy")
+    return True  # finally!
+
+
+def get_default_rpc_endpoints() -> Dict[int, List[str]]:
+    """
+    Fetches the default RPC endpoints for various chains
+    from the nucypher/chainlist repository.
+    """
+    LOGGER.debug("Fetching default RPC endpoints from remote chainlist")
+    response = requests.get(CHAINLIST_URL)
+    if response.status_code == 200:
+        return {
+            int(chain_id): endpoints for chain_id, endpoints in response.json().items()
+        }
+    else:
+        LOGGER.error(
+            f"Failed to fetch default RPC endpoints: {response.status_code} | {response.text}"
+        )
+        return {}
+
+
+def get_healthy_default_rpc_endpoints(chain_id: int) -> List[str]:
+    """Returns a list of healthy RPC endpoints for a given chain ID."""
+
+    healthy = list()
+    endpoints = get_default_rpc_endpoints()
+    chain_endpoints = endpoints.get(chain_id)
+
+    if not chain_endpoints:
+        LOGGER.error(f"No default RPC endpoints found for chain ID {chain_id}")
+        return healthy
+
+    for endpoint in chain_endpoints:
+        if rpc_endpoint_health_check(endpoint=endpoint):
+            healthy.append(endpoint)
```

Comment:
Can be done with list comprehension, but you can take it or leave it.

Reply:
✔️